crates/librqbit_core/src/torrent_metainfo.rs

```diff
@@ -62,8 +55,17 @@ impl<BufType> TorrentMetaV1<BufType> {
         }
         itertools::Either::Right(self.announce.iter())
     }
+
+    /// v1 called out, because v2 is a thing, and migration is a PITA.
+    pub fn v1_info_hash(&self) -> Id20 {
+        let mut h = sha1w::Sha1::new();
+        bencode::bencode_serialize_to_writer(&self.info, &mut h.write()).unwrap();
+        Id20::new(h.finish())
+    }
 }
 
+type RawInfo = bencode::BencodeValueOwned;
```

Comment:
Is it possible to make this generic over BufT?

Also if it's used once probably doesn't make sense to have a separate type for it.

i.e why not just

```
info: BencodeValue<BufT>
```

Reply:
I couldn't get that to work!